src/options.cpp

```diff
@@ -1320,6 +1322,31 @@ void show_options(bool ingame)
             g->init_ui();
             if( ingame ) {
                 g->refresh_all();
+
+                DebugLog( D_INFO, DC_ALL ) << "Loaded tileset: " << OPTIONS["TILES"].getValue();
+
+                tile_loading_report(termap, "Terrain", "");
+                tile_loading_report(furnmap, "Furniture", "");
+                tile_loading_report(Item_factory::Item_factory().get_all_itypes(), "Items", "");
```

Comment:
> I copied my idea for Item_factory from the existing code using MonsterGenerator. You say it should be empty, but it's not. The code as you commented on here does work.

It *compiles* fine, but it won't do what you want. Example:
The ChestHole tilset does not have a tile for "biollante_bud", why is it not appearing in the list of missing tiles in your example output?

The `MonsterGenerator` works differently, it calls the *static* function `MonsterGenerator::generator()` which returns the single instance of that class:
```C++
static MonsterGenerator &generator()
{
    static MonsterGenerator generator;

    return generator;
}
```
`MonsterGenerator::generator()` is a normal function. `Item_factory::Item_factory()` creates a *new* temporary object.